esp-hal/src/i2c/master/mod.rs

```diff
@@ -130,13 +130,21 @@ impl core::fmt::Display for Error {
 #[derive(Debug, Clone, Copy, PartialEq)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[non_exhaustive]
-pub enum ConfigError {}
+pub enum ConfigError {
+    /// Provided bus frequency is invalid for the current configuration.
+    InvalidFrequency,
```

Comment:
Next fun issue, now we have a precedent of `TimeoutInvalid`, should this be `FrequencyInvalid`?

Reply:
I guess, yes - that would be better